include/miral/miral/bounce_keys.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIRAL_BOUNCE_KEYS_H
+#define MIRAL_BOUNCE_KEYS_H
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+namespace mir { class Server; }
+
+namespace miral
+{
+/// Enables configuring bounce keys at runtime.
+/// \remark Since MirAL 5.3
+class BounceKeys
+{
+public:
+    explicit BounceKeys(bool enable_by_default);
```

Comment:
I'm not keen on using `bool` parameters like this. There's no context for readers of the code to interpret `true`/`false`. Example:

```c++
    BounceKeys bounce_keys{true};
```

Compare with alternative interface designs:

```c++
    auto bounce_keys = BounceKeys::enabled();
```

```c++
    BounceKeys bounce_keys{BounceKeys::enabled};
```

Reply:
I realise there's prior art in `HoverClick`, `MouseKeysConfig` and `SimulatedSecondaryClick` - sorry for not spotting this earlier